src/main/webapp/app/entities/text-block.model.ts

```diff
@@ -8,25 +8,64 @@ export enum TextBlockType {
 
 export class TextBlock {
     id?: string;
-    text?: string;
-    startIndex?: number;
-    endIndex?: number;
-    submission?: TextSubmission;
     type?: TextBlockType;
 
+    // The ID of the text block is computed as a hash of the submission ID, the start and end index and the text.
+    // We need to keep it up to date with the latest values of these properties. Therefore, we use setter properties to never forget to update the ID.
+    private _submissionId?: number;
+    private _startIndex?: number;
+    private _endIndex?: number;
+    private _text?: string;
+
+    set submissionId(value: number | undefined) {
+        this._submissionId = value;
+        this.computeId();
+    }
+
+    get submissionId(): number | undefined {
+        return this._submissionId;
+    }
+
+    set startIndex(value: number | undefined) {
+        this._startIndex = value;
+        this.computeId();
+    }
+
+    get startIndex(): number | undefined {
+        return this._startIndex;
+    }
+
+    set endIndex(value: number | undefined) {
+        this._endIndex = value;
+        this.computeId();
+    }
+
+    get endIndex(): number | undefined {
+        return this._endIndex;
+    }
+
+    set text(value: string | undefined) {
+        this._text = value;
+        this.computeId();
+    }
+
+    get text(): string | undefined {
+        return this._text;
+    }
+
     /**
-     * Identical with de.tum.in.www1.artemis.domain.TextBlock:computeId
+     * Computes the ID of the text block. The ID is a hash of the submission ID, the start and end index and the text.
      */
-    computeId(): void {
-        const submissionId = this.submission ? this.submission.id : 0;
+    private computeId(): void {
+        const submissionId = this.submissionId ?? 0;
         const idString = `${submissionId};${this.startIndex}-${this.endIndex};${this.text}`;
         this.id = sha1Hex(idString);
     }
 
     setTextFromSubmission(submission?: TextSubmission): void {
-        this.submission = submission || this.submission;
-        if (this.submission && !(this.startIndex === undefined || this.startIndex === null)) {
-            this.text = this.submission.text?.substring(this.startIndex, this.endIndex) || '';
+        this.submissionId ??= submission?.id;
+        if (submission && this.startIndex != undefined) {
```

Comment:
```suggestion
        if (submission && this.startIndex !== undefined) {

Reply:
I would keep it this way because it also matches `null`. The previous condition also made sure to also match `null` explicitly, so it might be sneaking in at some point in the client, although our guidelines say not to use it. Therefore, I would prefer to keep the check as-is, also matching `startIndex = null`.